tests/storage/mpathconf_test.py

```diff
@@ -102,3 +117,33 @@ def test_read_bad_blacklist(fake_conf):
     fake_conf.write(BAD_CONF)
     wwids = mpathconf.read_blacklist()
     assert wwids == {"wwid1", "wwid2", "wwid5"}
+
+
+@pytest.fixture
+def fake_mpath_conf(tmpdir, monkeypatch):
+    fake_conf = tmpdir.join("multipath.conf")
+    monkeypatch.setattr(mpathconf, "CONF_FILE", str(fake_conf))
+
+    def _write_data(data):
+        fake_conf.write(data)
+        return fake_conf
+
+    return _write_data
+
+
+@pytest.mark.parametrize("data", [
+    pytest.param(MPATH_CONF, id="current_tag"),
+    pytest.param(MPATH_CONF_PRIVATE, id="private_tag"),
+])
+def test_mpath_conf_valid(fake_mpath_conf, data):
+    fake_mpath_conf(data)
+    assert mpathconf.check_conf_revision()
+
+
+@pytest.mark.parametrize("data", [
+    pytest.param(MPATH_CONF_OLD_TAG, id="old_tag"),
+    pytest.param("", id="empty"),
+])
+def test_mpath_conf_invalid(fake_mpath_conf, data):
+    fake_mpath_conf(data)
+    assert not mpathconf.check_conf_revision()
```

Comment:
I don't think using pytest parameters is helpful in this case. It will be more clear to have one test per case with a good name explaining the use case and keep the test content in the test function.

For example:

    def test_missing_conf(fake_conf):
        assert mpathconf.check_conf_revision() == MultipathConf(
            revision=MISSING,
            private=False,
        )

    def test_empty_conf(fake_conf):
        fake_conf.write("")
        assert mpathconf.check_conf_revision() == MultipathConf(
            revision=MISSING,
            private=False,
        )

    def test_empty_conf(fake_conf):
        fake_conf.write("# VDSM REVISION 2.0\n# VDSM PRIVATE\ndata\n")
        assert mpathconf.check_conf_revision() == MultipathConf(
            revision=OLD,
            private=True,
        )

Using the actual values in the test is better than using the constants - these values can never change since they were already released, so we don't get anything by using the constants. Using constants helps when we can change the constant value and get all the relevant code use the new value transparently. This is actually the worst thing we can do here - if someone will change the constant in the module, the test will pass while the real code will fail in runtime trying to parse old values.

Using the actual values is also more clear - we can see the actual data in the test and we don't need to search for the value to understand the test.

Reply:
I'll change and use the actual values instead of the constants. Got it.

Regarding the pytest parameters, I saw that the lvmconf_test was doing something similar and I liked it:
```
@pytest.mark.parametrize("data", [
    pytest.param(CONF, id="example"),
    pytest.param("", id="empty"),
])
def test_modify(tmpdir, data):
```
But with the new approach for what the function returns is true that it is a bit too complicated to keep it like that, so I'll change that too.